lib/pg_eventstore/commands/regular_stream_read_paginated.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module PgEventstore
+  module Commands
+    # @!visibility private
+    class RegularStreamReadPaginated < AbstractCommand
+      # @see PgEventstore::Commands::Read for docs
+      def call(stream, options: {})
+        revision = calc_initial_revision(stream, options)
+        Enumerator.new do |yielder|
+          loop do
+            events = read_cmd.call(stream, options: options.merge(from_revision: revision))
+            yielder << events if events.any?
+            raise StopIteration if end_reached?(events, options[:max_count])
```

Comment:
I wonder if you do not get more events in a case: 
If we assume we have 2k events the bach size is 1k  and let's say I want max_count 1.5k of events. Didn't you get any here 2k? because in line 13 you will always read in batch which is 1k.

I mean about `PgEventstore.client.read_paginated(...).take(1500).each {....}`

Nice approach btw.